### ISSUE TITLE
URL normalization

### ISSUE THREAD

**Issue by owenson** [gocolly/colly#361]
Colly does not currently appear to do any URL normalization. For example querystring parameters need to be reordered in alphabetical order, host lowercased, etc.

See https://github.com/PuerkitoBio/purell

**Comment by asciimoo** [comment]
Why do parameters need to be reordered?

**Comment by owenson** [comment]
/test?a=1&b=2 and /test?b=1&a=2 point to the same thing

**Comment by asciimoo** [comment]
@owenson this depends on the webapp, it can distinguish these cases and return different results for  the two urls.
Also in the first url `a=1` and `b=2` but in the second `a=2` and `b=1`, i assume this is just a typo.

**Comment by owenson** [comment]
Yes it was a typo. I've yet to see a web app which cares about order, or indeed any Web frameworks that let you get that info. Most crawlers will normalise the query parameters in this way. Nutch does it, scrapy does it, etc.

**Comment by asciimoo** [comment]
I'm open to adding an option to normalize url. Would you like to work on this?

**Comment by owenson** [comment]
I've used this in the past.

https://github.com/PuerkitoBio/purell

**Comment by asciimoo** [comment]
Great, thanks!
